// utils/variables.js
const colors = {
  // Gray Colors
  silver:               '#e1e1e2',
  gray:                 '#E6E6E6',
  tags:                 '#D2D3D3',
  gray_button:          '#cccccc',
  digit_radio:          '#E2E2E2',
  light_gray:           '#CAD0D5',
  dark_gray:            '#798B99',
  placeholders:         '#B0A7AF',
  middle_gray:          '#8D8A8D',
  nav_account:          '#554F54',
  charcoal_grey:        '#35383e',
  invention_navigation: '#423C41',
  login_background:     '#2F2A2F',
  form_description:     '#978F96',

  // White
  clear_white:          '#ffffff',
  background_dashboard: '#FBFBFF',
  gray_white:           '#f0f0f0',
  white:                '#f7f7f7',
  darken_white:         '#F8F8F8',
  dark_white:           '#F9F9F9',

  // Red
  coral_pink:           '#ff5a5f',
  coral_pink_darken:    '#CC474C',
  brownish_red:         '#7f2c2c',

  // Green
  greenish_teal:        '#3fd485',
  greenish_teal_darken: '#49B57C',
  pine:                 '#376e50',

  // Brown
  sepia:                '#7f532c',
  dirt_brown:           '#7a4631',
  purple_brown:         '#30292f',

  // Orange
  pale_orange:          '#ffa759',
  pale_orange_darken:   '#D3935B',

  // Without Category
  bright_lavender:      '#c259ff',

  // Form Buttons
  button_green:         '#3fd485',
  button_red:           '#ff5a5f',
  button_orange:        '#ffa759',
  button_gray:          '#B0A7AF',
  button_lightgray:     '#CCCCCC'
}

export default colors
